src/Build/Graph/ProjectGraph.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Build.Exceptions;
+
+namespace Microsoft.Build.Graph
+{
+    /// <summary>
+    /// Represents a graph of evaluated projects.
+    /// </summary>
+    internal class ProjectGraph
+    {
+        /// <summary>
+        /// Constructs a graph starting from the given project file.
+        /// </summary>
+        /// <param name="projectFile">The project file</param>
+        /// <exception cref="InvalidProjectFileException">If the evaluation of any project in the graph fails.</exception>
+        public ProjectGraph(string projectFile) { }
```

Comment:
I think I'd prefer the name `entryProject` for this parameter. Especially since in VS land it'll likely often be a solution.

Reply:
I like `entryProjectFile` (`entryProject` maybe for an overload that takes a `Project`), but I was hoping that for VS they'd just use the overload that takes multiple projects and not feed us a sln file here. They already know all the projects; no need to force MSBuild to parse the sln all over again.

Reply:
I agree that inside VS it makes sense for them to hand us the list of projects. I was thinking mostly of command-line builds of solutions (as opposed to traversal-project-driven builds). Those are super common outside Microsoft.

Reply:
Poop, you're right. Although potentially we could do the metaproj thing and feed that to here instead of the sln itself, but that's an impl detail we can get to when the time comes :)